lib/rust/ensogl/core/src/display/shape/primitive/glsl/shape.glsl

```diff
@@ -315,21 +333,27 @@ Shape debug_shape (BoundSdf bound_sdf) {
 Shape resample (Shape s, float multiplier) {
     Id id = s.id;
     BoundSdf sdf = resample(s.sdf, multiplier);
-    s.color.repr.raw.a /= s.alpha;
+    if (input_display_mode != DISPLAY_MODE_CACHED_SHAPES_TEXTURE) {
+        s.color.repr.raw.a /= s.alpha;
+    }
     return shape(id, sdf, s.color);
 }
 
 Shape pixel_snap (Shape s) {
     Id id = s.id;
     BoundSdf sdf = pixel_snap(s.sdf);
-    s.color.repr.raw.a /= s.alpha;
+    if (input_display_mode != DISPLAY_MODE_CACHED_SHAPES_TEXTURE) {
+        s.color.repr.raw.a /= s.alpha;
+    }
     return shape(id, sdf, s.color);
 }
 
 Shape grow (Shape s, float value) {
     Id id = s.id;
     BoundSdf sdf = grow(s.sdf,value);
-    s.color.repr.raw.a /= s.alpha;
+    if (input_display_mode != DISPLAY_MODE_CACHED_SHAPES_TEXTURE) {
+        s.color.repr.raw.a /= s.alpha;
+    }
     return shape(id, sdf, s.color);
```

Comment:
these lines need a good explanation why we are doing it this way. The explanation can be put in one place and these places should link to it. The explanation should describe the difference in the process of rendering in this mode to explain why these changes are needed. I understand it, but I believe that a reader who was not involved in a process of creation of this would not be able to follow up with the logic here.